src/main/java/uk/ac/ox/oxfish/biology/growers/FadAwareLogisticGrower.java

```diff
@@ -0,0 +1,211 @@
+        final double currentBiomass =
+            allBiologies(fishState).mapToDouble(biology -> biology.getBiomass(species)).sum();
+
+        // if we are in "Schaefer mode", the biomass we'll feed to the logistic growth
+        // function is the biomass that was memorized at the beginning of the year,
+        // otherwise it's just the current biomass.
+        final double biomassToUse = memorizer.map(m -> m.memorizedBiomass).orElse(currentBiomass);
+
+        // we call the logistic function (r  * biomassToUse * (1 - biomassToUse / K))
+        // to get the new biomass resulting from growth and recruitment
+        final double newBiomass = logisticRecruitment(biomassToUse, totalCapacity, malthusianParameter);
+
+        // we calculate how much space we have left in the ocean to put new biomass
+        final double availableCapacity = totalCapacity - currentBiomass;
+
+        // the biomass to allocate is the sum of the new biomass and the biomass lost by FADs drifting out,
+        // while making sure we won't be exceeding the total carrying capacity of the ocean tiles
+        final double biomassLostByFads = biomassLostAccumulator.map(Accumulator::get).orElse(0.0);
+        final double biomassToAllocate = Math.min(newBiomass + biomassLostByFads, availableCapacity);
+
+        // the biomass to allocate should not be negative, barring tiny floating point errors
+        assert biomassToAllocate >= -EPSILON : "biomassToAllocate: " + biomassToAllocate;
+
+        if (biomassToAllocate > EPSILON) {
+            // if we have some new biomass to allocate, we distribute it among the sea tiles (not the FADs)
+            allocateBiomass(biomassToAllocate, fishState.getRandom());
+            final String columnName = fishState.getSpecies().get(species.getIndex()) + " Recruitment";
+            fishState.getYearlyCounter().count(columnName, biomassToAllocate);
+        }
+    }
+
+    private Stream<BiomassLocalBiology> allBiologies(final FishState fishState) {
+        return concat(fadBiologies(fishState), seaTileBiologies.stream());
+    }
+
+    private void allocateBiomass(final double biomassToAllocate, MersenneTwisterFast rng) {
+        if (distributionalWeight > 0)
+            allocateBiomassProportionally(
+                seaTileBiologies,
+                biomassToAllocate,
+                species.getIndex(),
+                distributionalWeight
+            );
+        else
+            allocateBiomassAtRandom(
+                seaTileBiologies,
+                biomassToAllocate,
+                rng,
+                species.getIndex()
+            );
+    }
+
+    @NotNull @SuppressWarnings("UnstableApiUsage")
+    private Stream<BiomassLocalBiology> fadBiologies(final FishState fishState) {
+        return stream(Optional.ofNullable(fishState.getFadMap()))
+            .flatMap(FadMap::allFads)
+            .map(Fad::getBiology);
+    }
+
+    private class Memorizer implements Steppable, Startable {
```

Comment:
do we need to stop this?

Reply:
Not unless we stop the grower itself, and I'm not sure we'll ever need to do that, but I must admit I always waiver about writing proper stoppers or not. Do you have a policy about that? Should every steppable also be stoppable?

Reply:
I feel stopping is important for things that do stop (strategies of agents, for example). So most of the time don't worry about it.

For grower the point may be the following: imagine you need to run a climate change scenario and climate change is modelled as a low steepness. Do you need to stop anything in this case?
Maybe you can cross that bridge when you get there